awsxrayproxy: TestFactory_CreateExtension failed

Test failed here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5694/checks?check_run_id=3881454973

```
make[2]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/awsxrayproxy'
go test -race -timeout 300s --tags="" ./...
--- FAIL: TestFactory_CreateExtension (0.00s)
    factory_test.go:71: 
        	Error Trace:	factory_test.go:71
        	Error:      	Received unexpected error:
        	            	Post "http://127.0.0.1:41973/GetSamplingRules": dial tcp 127.0.0.1:41973: connect: connection refused
        	Test:       	TestFactory_CreateExtension
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xfed37e]

goroutine 6 [running]:
testing.tRunner.func1.2({0x1094840, 0x213d380})
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1209 +0x36c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1212 +0x3b6
panic({0x1094840, 0x213d380})
	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/panic.go:1047 +0x266
github.com/open-telemetry/opentelemetry-collector-contrib/extension/awsxrayproxy.TestFactory_CreateExtension(0x0)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/awsxrayproxy/factory_test.go:74 +0x79e
testing.tRunner(0xc000001a00, 0x11b6ad8)
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1259 +0x230
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1306 +0x727
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/extension/awsxrayproxy	0.044s
FAIL
make[2]: *** [../../Makefile.Common:45: test] Error 1
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/awsxrayproxy'
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
make[1]: *** [Makefile:148: for-all-target-./extension/awsxrayproxy] Error 2
make: *** [Makefile:68: gotest] Error 2
Error: Process completed with exit code 2.
```

--- a/extension/awsxrayproxy/extension.go
+++ b/extension/awsxrayproxy/extension.go
@@ -16,6 +16,7 @@ package awsxrayproxy
 
 import (
 	"context"
+	"net/http"
 
 	"go.opentelemetry.io/collector/component"
 	"go.uber.org/zap"
@@ -32,7 +33,13 @@ type xrayProxy struct {
 var _ component.Extension = (*xrayProxy)(nil)
 
 func (x xrayProxy) Start(ctx context.Context, host component.Host) error {
-	go x.server.ListenAndServe()
+	go func() {
+		if err := x.server.ListenAndServe(); err != nil {
+			if err != http.ErrServerClosed {
+				host.ReportFatalError(err)
+			}
+		}
+	}()
 	x.logger.Info("X-Ray proxy server started on " + x.config.ProxyConfig.Endpoint)
 	return nil
 }

--- a/extension/awsxrayproxy/factory_test.go
+++ b/extension/awsxrayproxy/factory_test.go
@@ -16,12 +16,15 @@ package awsxrayproxy
 
 import (
 	"context"
+	"fmt"
 	"net/http"
+	"net/http/httptest"
 	"os"
 	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/confignet"
@@ -46,8 +49,22 @@ func TestFactory_CreateDefaultConfig(t *testing.T) {
 }
 
 func TestFactory_CreateExtension(t *testing.T) {
+	backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		auth := r.Header.Get("Authorization")
+		// Verify a signature was added, indicating the reverse proxy is doing its job.
+		if !strings.HasPrefix(auth, "AWS4-HMAC-SHA256") {
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprintln(w, "No signature")
+			return
+		}
+		w.Header().Set("Test", "Passed")
+		fmt.Fprintln(w, "OK")
+	}))
+	defer backend.Close()
+
 	cfg := createDefaultConfig().(*Config)
 	address := testutil.GetAvailableLocalAddress(t)
+	cfg.ProxyConfig.AWSEndpoint = backend.URL
 	cfg.ProxyConfig.TCPAddr.Endpoint = address
 	cfg.ProxyConfig.Region = "us-east-2"
 
@@ -60,7 +77,8 @@ func TestFactory_CreateExtension(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, ext)
 
-	err = ext.Start(ctx, componenttest.NewNopHost())
+	mh := newAssertNoErrorHost(t)
+	err = ext.Start(ctx, mh)
 	assert.NoError(t, err)
 
 	resp, err := http.Post(
@@ -70,9 +88,27 @@ func TestFactory_CreateExtension(t *testing.T) {
 
 	assert.NoError(t, err)
 
-	// The request was proxied and has standard AWS headers.
-	assert.NotEmpty(t, resp.Header.Get("X-Amzn-Requestid"))
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	assert.Equal(t, "Passed", resp.Header.Get("Test"))
 
 	err = ext.Shutdown(ctx)
 	assert.NoError(t, err)
+}
+
+// assertNoErrorHost implements a component.Host that asserts that there were no errors.
+type assertNoErrorHost struct {
+	component.Host
+	*testing.T
+}
+
+// newAssertNoErrorHost returns a new instance of assertNoErrorHost.
+func newAssertNoErrorHost(t *testing.T) component.Host {
+	return &assertNoErrorHost{
+		Host: componenttest.NewNopHost(),
+		T:    t,
+	}
+}
+
+func (aneh *assertNoErrorHost) ReportFatalError(err error) {
+	assert.NoError(aneh, err)
 }

